app/src/main/java/woowacourse/shopping/data/mapper/Mapper.kt

```
@@ -57,3 +68,34 @@ fun ProductResponse.toProduct() =
+            discount = discount,
+            minimumAmount = minimumAmount,
+            buyQuantity = buyQuantity,
+            getQuantity = getQuantity,
+            availableTime = timeRange,
+        )
+    return when (discountType) {
+        "fixed" -> FixedCoupon(couponInfo, CouponType.FIXED)
+        "percentage" -> PercentageCoupon(couponInfo, CouponType.PERCENTAGE)
+        "buyXgetY" -> BuyXGetYFreeCoupon(couponInfo, CouponType.BUY_X_GET_Y)
+        "freeShipping" -> FreeShippingCoupon(couponInfo, CouponType.FREE_SHIPPING)
+        else -> throw IllegalStateException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)